source/leuven/services/openondemand.rst

```diff
@@ -191,136 +241,297 @@ of using any of the other partitions. There are however some limitations on the
-environment::
+   If you use JupyterLab as explained above, and you need to reproduce your numerical results 
+   at any time in the future, make sure you stick to choosing the same 'Toolchain version'.
+
+If the module-based environment explained above does not provide all packages that you need, then
+it is recommended to manage your custom-made R or Python environments by creating custom Jupyter kernels.
+
+Custom Conda environments
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have not installed Conda in your account yet, please refer to the
+:ref:`install Miniconda <install_miniconda_python>` page.
+
+To create a custom kernel, first create a  :ref:`Python <create_python_conda_env>` or
+:ref:`R <create_r_conda_env>` Conda environment. The second step consists of effectively
+creating the kernel in your ``$VSC_HOME/.local/share`` folder (which is a defalt value for the
+``$XDG_DATA_HOME`` environment variable), as Jupyter will look for kernels in this location.
+The following commands should be excecuted from a shell (e.g. using 'Login Server Shell Access'), and only need
+to be done once for the set-up of each new kernel.
+If you already have an existing Python kernel, but your JupyterLab session freezes/craches when choosing your
+old kernel, you also need to repeat the following steps only once.
+This starts with activating your Conda environment::
 
       source activate <env_name>
 
-For Python you will need the ``ipykernel`` package installed in your conda environment. Then you create the kernel as follows::
+For Python you will need the ``ipykernel`` package installed in your Conda environment.
+The minimum supported version for Python is 3.7 and for ``ipykernel`` package is 6.19.2::
+
+      conda install ipykernel
+
+Then you create the kernel as follows::
 
-      python -m ipykernel install --user --name '<env_name>' --display-name '<kernel_name>'
+      python -m ipykernel install --user --env PYTHONPATH "" --name '<env_name>' --display-name '<kernel_name>'
 
-For R, you need both the ``jupyter_client`` and the ``irkernel`` package installed. With the following command you can create the kernel::
+
+Custom virtual environments for Python
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One may create a customized kernel starting from a Python virtual environment, using one of Python modules.
+Eventhough this approach can have advantages, but we do not recommend it, due to the fact that Python
+virtual environments are architecture specific.
+Hence, on our current Tier-2 machines, this approach needs to be handled with care.
+If you are interested in this approach, follow these steps to create and use your kernel:
+
+- Pick a specific Python from a specific 'Toolchain version', e.g. ``Python/3.11.3-GCCcore-12.3.0``
+  from ``2023a``
+- Choose a specific architecture, e.g. Sapphire Rapids nodes on wICE
+- Start an :ref:`Interactive Shell<interactive_shell>` on the targeted architecture,
+  and execute the following:
```

Comment:
I think it is best to have 'basic' and 'advanced' scenarios:
- What is currently written here would correspond to the 'advanced' one, where the user anticipates that architecture-dependent installations will be needed. This is normally only needed under certain circumstances (e.g. when installing some packages requires compiling extensions from source).
- So probably best to start with a simpler 'basic' scenario (before showing the more advanced case) where the architecture dependence is not explicitly considered, e.g.:
```bash
     python -m venv ${VSC_DATA}/venv_science_2023a
     (etcetera)
```

Reply:
The text in the JupyterLab secion has already grown, and I prefer to keep it as least complex as possible. Splitting a sub-section into two scenarios just adds to the length of the text and its complexity, which I strongly like to avoid here.
Users who want to create a kernel out of Python venv are already advanced users, so they shall follow only one set of instructions. This is preferable on the support-side.
For that reason, I prefer to keep only one scenario here.